x-pack/plugins/features/server/oss_features.ts

```diff
@@ -20,21 +20,21 @@ export const buildOSSFeatures = ({ savedObjectTypes, includeTimelion }: BuildOSS
       }),
       order: 100,
       icon: 'discoverApp',
-      navLinkId: 'kibana:discover',
-      app: ['kibana'],
+      navLinkId: 'discover',
+      app: ['discover', 'kibana'],
       catalogue: ['discover'],
       privileges: {
         all: {
-          app: ['kibana'],
+          app: ['discover', 'kibana'],
           catalogue: ['discover'],
           savedObject: {
-            all: ['search', 'query'],
-            read: ['index-pattern'],
+            all: ['search', 'query', 'index-pattern'],
```

Comment:
Ah Good Point, I fixed that as a by-product but I can also revert here and move it into a separate PR if you want. Discover does write index pattern objects because it changes the “popularity” of fields.

Reply:
> Ah Good Point, I fixed that as a by-product but I can also revert here and move it into a separate PR if you want. Discover does write index pattern objects because it changes the “popularity” of fields.

Interesting. Do you know if Discover properly handles "readonly" users today for popularity counts, or does it fail? I know there are some index-pattern related issues for read-only users today (such as https://github.com/elastic/kibana/issues/18019)

Reply:
I don’t know but I can check and open an issue if it doesn’t handle read rights correctly. For “all” rights on discover is still makes sense to grant write access to index patterns IMHO because it’s part of the feature. What do you think?

Reply:
> I don’t know but I can check and open an issue if it doesn’t handle read rights correctly. For “all” rights on discover is still makes sense to grant write access to index patterns IMHO because it’s part of the feature. What do you think?

Yeah that makes sense to me. I don't mind if you leave it as part of this PR if that's easier. I just wanted to understand the reason for the change